### PR TITLE
go/parser: support type alias

### PR DESCRIPTION
**Pull request by gofly**
Support finding symbal in the package which has type alias defination in.

### CODE REVIEW

File: go/parser/parser.go [hunk 1]
```diff
@@ -2028,6 +2028,9 @@ func parseTypeSpec(p *parser, doc *ast.CommentGroup, decl *ast.GenDecl, _ int) a
 	// containing block.
 	// (Global identifiers are resolved in a separate phase after parsing.)
 	spec := &ast.TypeSpec{doc, ident, nil, p.lineComment}
+	if p.tok == token.ASSIGN {
```
Comment (uudashr) [inline go/parser/parser.go]: Is it work for Go version below 1.9?

**Comment by rogpeppe** [comment]
Unfortunately this doesn't work - type aliases need to retain all the methods of the original type.

For example, in the code below, doing a godef on the "Encode" in `x.Encode` does not find any definition, but it should.

    package main
    import "encoding/json"
    func main() {
	type foo = json.Encoder
	x := new(foo)
	_ = x.Encode
    }

**Comment by jackbearheart** [comment]
It might be nice to merge this with a FIXME at least for the short term. With this patch the parser will regard a package with type aliases as parseable, but in master if you call godef on a type that is even in the same package as a type alias, it won't work at all. For me godef is totally broken w/o this patch when I use type aliases.

I'd offer to code up the full fix but I'd have to learn more godef internals...

**Comment by rogpeppe** [comment]
Superceded by #77

**Pull request closed without merging**